ethereum/rpc/eth_api.go

```diff
@@ -678,12 +678,14 @@ func (e *PublicEthAPI) GetTransactionByHash(hash common.Hash) (*rpctypes.RPCTran
 func (e *PublicEthAPI) GetTransactionByBlockHashAndIndex(hash common.Hash, idx hexutil.Uint) (*rpctypes.RPCTransaction, error) {
 	e.logger.Debugln("eth_getTransactionByHashAndIndex", "hash", hash.Hex(), "index", idx)
 
-	resp, err := e.queryClient.TxReceiptsByBlockHash(e.ctx, &evmtypes.QueryTxReceiptsByBlockHashRequest{
-		Hash: hash.Hex(),
-	})
+	header, err := e.backend.HeaderByHash(hash)
 	if err != nil {
-		err = errors.Wrap(err, "failed to query tx receipts by block hash")
-		return nil, err
+		return nil, errors.Wrapf(err, "failed retrieve block from hash")
+	}
+
+	resp, err := e.queryClient.TxReceiptsByBlockHeight(rpctypes.ContextWithHeight(header.Number.Int64()), &evmtypes.QueryTxReceiptsByBlockHeightRequest{})
```

Comment:
because you can retrieve the block by hash on the Tendermint RPC client, but you can't access it from the store. With this approach, we can get rid of the block hash -> block height map on the KVStore